### ISSUE TITLE
NVDA updating fails with Windows Protected Folders enabled

### ISSUE THREAD

**Issue by ABuffEr** [nvaccess/nvda#7735]
Hi,
I use NVDA master installed on Windows 10 1709 (Fall Creators Update), and I enabled Protected Folders, a new anti-ransomware feature of this Windows version, more info here:
https://docs.microsoft.com/en-us/windows/threat-protection/windows-defender-exploit-guard/customize-controlled-folders-exploit-guard
I also customized Temp folder path, I think that is irrelevant, but I cannot test it, sorry.
Now...

### Steps to reproduce:
1. Open Windows Defender Security Center, from start menu, navigate to protection settings and activate Protected Folders;
2. Restart Windows to fully enable the feature;
3. Try to update NVDA.

### Expected behavior:
NVDA updates its files and asks to restart.

### Actual behavior:
NVDA says the update failed, and a Windows notification alerts that C:\Temp\...\nvda_slave.exe cannot write on %common_desktop%\.

### NVDA log:
ERROR - RPC process 2404 (nvda_slave.exe) (21:12:31.595):
__main__.main:
slave error
Traceback (most recent call last):
  File "nvda_slave.pyw", line 42, in main
  File "installer.pyc", line 413, in install
  File "installer.pyc", line 226, in registerInstallation
  File "installer.pyc", line 56, in createShortcut
  File "comtypes\client\lazybind.pyc", line 163, in caller
  File "comtypes\automation.pyc", line 664, in _invoke
COMError: (-2147352567, 'Eccezione.', (None, None, None, 0, None))
ERROR - gui.installerGui.doInstall (21:12:31.688):
Installation failed: 1

### Final note
Note that:
*  unfortunately, Protected Folders feature protects as default public and user desktop directories, and these are not removable from protected folders list;
*  the whitelisted apps are checked also for path, so random generated .tmp path for nvda_slave.exe is not useful here.

The easiest workaround is, obviously, to disable the feature, update NVDA, and re-enable... if user has the rights to do it.
In alternative, can you arrange the desktop shortcut creation/update process in a way that it is executed from a non-random path?

**Comment by feerrenrut** [comment]
Could you update the description to provide the NVDA version you are using?

This is a duplicate of #7696. So I will close this issue, however please feel free to comment on it.

Which has a fix incubating (available in `next` builds), however, there is a known issue remaining, the desktop shortcut can not be created when Protected Folders is active. 

Please can you try updating again, and untick the create desktop shortcut option in the installer.

**Comment by ABuffEr** [comment]
Ops, I searched without finding the previous issue...
Updated with master version, re-installed (same version) unticking the desktop checkbox, and it works fine (but Windows notification appears in any case).
About remaining issue of shortcut creation (the first time, I suppose), I'm thinking... would it be possible to create a shortcut as pin in taskbar? I see two folders in %APPDATA%\Microsoft\Internet Explorer\Quick Launch\User Pinned, and that is, TaskBar and ImplicitAppShortcuts. I don't know the difference, but copying the NVDA link from desktop in these locations (and changing the associated keyboard shortcut for testing), these work well.

**Comment by Brian1Gaff** [comment]
One thought of course is that if the actual software is installed, that even 
though the shortcut is not installed for the first time, it is instead 
created somewhere easy to get to and leave the temp copy running or start 
the installed copy and present a message about copying the shortcut 
manually.

Tonsure if that would be allowed but most people  seem to want to put 
shortcuts and files on their desctop at least for temp storage this feature 
being too draconian is going to get on peoples nerves big time!

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----
